### PR TITLE
feat: added cmd to encrypt tss keyshare file

### PR DESCRIPTION
**Pull request by kevinssgh**
added cmd to encrypt tss keyshare file, allowing empty tss password for backward compatibility.

# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes: <PD-XXXX>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: cmd/zetaclientd/encrypt_tss.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	filePath  string
+}
+
+var tssArgs = TSSArgs{}
+
+func init() {
+	RootCmd.AddCommand(encTssCmd)
+
+	encTssCmd.Flags().StringVar(&tssArgs.secretKey, "secret", "", "tss-encrpyt --secret p@$$w0rd")
+	encTssCmd.Flags().StringVar(&tssArgs.filePath, "filepath", "", "tss-encrpyt --filepath ./file.json")
```
Comment (lumtis) [inline cmd/zetaclientd/encrypt_tss.go:32]: Why not using positional argument instead?

**Review by lumtis — COMMENTED** [review]
Need to address lint issues

File: cmd/zetaclientd/encrypt_tss.go (hunk 1 above)
Comment (kevinssgh) [reply cmd/zetaclientd/encrypt_tss.go]: I think it is more clear what the arguments are like this but it does work either way. I can change it to positional if necessary.

Comment (lumtis) [reply cmd/zetaclientd/encrypt_tss.go]: As a general rule, if it is required, positional is more adapted

File: cmd/zetaclientd/encrypt_tss.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+var encTssCmd = &cobra.Command{
+	Use:   "tss-encrypt [file-path] [secret-key]",
+	Short: "Utility command to encrypt existing tss key-share file",
+	Args:  cobra.ExactArgs(2),
+	RunE:  EncryptTSSFile,
+}
+
+type TSSArgs struct {
+	secretKey string
+	filePath  string
```
Comment (lumtis) [inline cmd/zetaclientd/encrypt_tss.go:25]: It doesn't seem this structure is really needed

File: cmd/zetaclientd/start.go [hunk 3]
```diff
@@ -333,9 +333,9 @@ func promptPasswords() (string, string, error) {
 		return "", "", err
 	}
 
-	if TSSKeyPass == "" {
-		return "", "", errors.New("tss password is required to start zetaclient")
-	}
+	//trim delimiters
+	hotKeyPass = strings.TrimSuffix(hotKeyPass, "\n")
```
Comment (ws4charlie) [inline cmd/zetaclientd/start.go:337]: these two lines are to allow empty pass for backwards compatibility? just to confirm

**Review by ws4charlie — APPROVED** [review]
looks good

File: cmd/zetaclientd/encrypt_tss.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+		return err
+	}
+	// Generating random nonce
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err := io.ReadFull(rand.Reader, nonce); err != nil {
+		return err
+	}
+
+	cipherText := gcm.Seal(nonce, nonce, data, nil)
+	return os.WriteFile(tssArgs.filePath, cipherText, 0o600)
```
Comment (ws4charlie) [inline cmd/zetaclientd/encrypt_tss.go:64]: will this overwrite contents of existing file?

File: cmd/zetaclientd/start.go (hunk 3 above)
Comment (kevinssgh) [reply cmd/zetaclientd/start.go:337]: Yes, the read function includes the line feed character so we need to remove them

File: cmd/zetaclientd/encrypt_tss.go (hunk 4 above)
Comment (kevinssgh) [reply cmd/zetaclientd/encrypt_tss.go]: Yes, if they want to do a backup it needs to be done before.

**Pull request merged**